nidirect_common/nidirect_common.module

```diff
@@ -302,20 +308,60 @@ function nidirect_common_validate_publication_date(&$form, FormStateInterface $f
   // Clear the form errors.
   $form_state->clearErrors();
 
+  $date_fields = ['field_published_date',
+    'field_last_review_date',
+    'field_next_review_date',
+    'field_cwp_payments_period',
+  ];
+
   // Now loop through and re-apply all form error messages that
-  // don't involve 'field_published_date'.
+  // don't involve the specified date fields.
   foreach ($form_errors as $name => $error_message) {
-    if (substr($name, 0, 20) != 'field_published_date') {
-      $form_state->setErrorByName($name, $error_message);
+    if (preg_match('/^[a-z_]*/', $name, $matches)) {
```

Comment:
`^[a-z_]*` should match any lowercase alphanumeric character at the start of the string. The `*` quantifier means zero or more of the last matching group. Does that mean this will then always be true? I'd normally consider `+` which is one or more of the last group.

Reply:
There should always be something there, so have changed this to a '+' as suggested